pkg/kv/kvserver/kvstorage/init.go

```diff
@@ -347,6 +375,20 @@ func LoadAndReconcileReplicas(ctx context.Context, eng storage.Engine) (*EngineR
 					ident.StoreID, desc)
 			}
 
+			sp := desc.RSpan().AsRawSpanWithNoLocals()
```

Comment:
Could you add a comment above saying what this block does? It looks to me like it checks that all range descriptors are non-overlapping.

Do you need the interval tree here? If `IterateRangeDescriptorsFromDisk` returns descriptors ordered by the first key (and the `INVARIANT` also says that begin/end keys are verified to match the key at which the descriptor is located), you only need to check that `current.Key` >= `prev.EndKey`. You can also double-check that `current.Key` > `prev.Key` (unless you trust the iteration func, in which case it's sufficient to make a comment).

Side: this seems out of place in this commit/PR, would prefer to have reviewed it separately.

Reply:
I brought it back, since it's a one-liner now. I had it in a separate commit but the whitespace changes made it more trouble than is worth it.